go-controller/pkg/ovn/controller/services/utils.go

```diff
@@ -3,388 +3,116 @@ package services
-				klog.Errorf("Gateway router %s does not have load balancer (%v)",
-					gatewayRouter, err)
-				continue
-			}
-		}
-		physicalIPs, err := gateway.GetGatewayPhysicalIPs(gatewayRouter)
-		if err != nil {
-			klog.Errorf("Gateway router %s does not have physical ip (%v)", gatewayRouter, err)
-			continue
+	// Find any legacy LBs with these vips
+	for _, lb := range legacyLBs {
+		r := ovnlb.DeleteVIPEntry{
+			LBUUID: lb.UUID,
 		}
 
-		// Get node-local endpoints only
-		if nodeLocal && config.Gateway.Mode == config.GatewayModeShared {
-			klog.V(5).Infof("ExternalTrafficPolicy=Local only getting local endpoints for node %s", util.GetWorkerFromGatewayRouter(gatewayRouter))
-			targetIPs = eps.LocalToNode(gatewayRouter).IPs()
+		proto := v1.Protocol(strings.ToUpper(lb.Protocol))
+		for _, vip := range vipPortsPerProtocol[proto].List() {
+			if _, ok := lb.VIPs[vip]; ok {
+				r.VIPs = append(r.VIPs, vip)
+			}
 		}
 
-		var newTargets []string
-
-		if config.Gateway.Mode == config.GatewayModeShared {
-			// If self ip is in target list, we need to use special IP to allow hairpin back to host
-			newTargets = util.UpdateIPsSlice(targetIPs, physicalIPs, []string{types.V4HostMasqueradeIP, types.V6HostMasqueradeIP})
-		} else {
-			newTargets = targetIPs
+		if len(r.VIPs) > 0 {
+			toRemove = append(toRemove, r)
 		}
-		lbConfig = append(lbConfig, loadbalancer.Entry{
-			LoadBalancer: gatewayLB,
-			SourceIPS:    svcIPs,
-			SourcePort:   sourcePort,
-			TargetIPs:    newTargets,
-			TargetPort:   eps.Port,
-		})
+	}
 
-		if config.Gateway.Mode == config.GatewayModeShared {
-			workerNode := util.GetWorkerFromGatewayRouter(gatewayRouter)
-			workerLB, err := loadbalancer.GetWorkerLoadBalancer(workerNode, protocol)
-			if err != nil {
-				klog.Errorf("Worker switch %s does not have load balancer (%v)", workerNode, err)
-				return err
-			}
-			lbConfig = append(lbConfig, loadbalancer.Entry{
-				LoadBalancer: workerLB,
-				SourceIPS:    svcIPs,
-				SourcePort:   sourcePort,
-				TargetIPs:    targetIPs,
-				TargetPort:   eps.Port,
-			})
-		}
+	if err := ovnlb.DeleteLoadBalancerVIPs(toRemove); err != nil {
+		return fmt.Errorf("failed to delete vip(s) from legacy load balancers: %w", err)
 	}
-	return loadbalancer.BundleCreateLoadBalancerVIPs(lbConfig)
+	return nil
 }
 
-// createPerNodePhysicalVIPs adds load balancers on a per node basis for GR and worker switch LBs using physical IPs
-func createPerNodePhysicalVIPs(isIPv6 bool, protocol v1.Protocol, sourcePort int32, eps util.LbEndpoints, nodeLocal bool) error {
-	var gatewayLB string
-	targetIPs := eps.IPs()
-	klog.V(5).Infof("Creating Node VIPs - %s, %d, [%v], %d %v", protocol, sourcePort, eps.IPs(), eps.Port, nodeLocal)
-	// Each gateway has a separate load-balancer for N/S traffic
-	gatewayRouters, _, err := gateway.GetOvnGateways()
+// getLegacyLBs returns all of the "legacy" non-per-Service load balancers
+// This is any load balancer with one of the following external ID keys
```

Comment:
+ the legacy ETP LB 

// - <PROTO>_lb_gateway_router_local

Reply:
Oh duh, very good catch!

Reply:
No, I was right the first time: we only match on key, not value, so we don't need to worry.